cluster-operator/src/main/java/io/strimzi/operator/cluster/crd/model/AbstractSsLike.java

```diff
@@ -0,0 +1,131 @@
+import io.strimzi.crdgenerator.annotations.Minimum;
+
+import java.util.Map;
+
+/**
+ * A baseclass for {@link Kafka} and {@link Zookeeper} with all their common properties.
+ */
+public abstract class AbstractSsLike {
+
+    protected int replicas;
+
+    protected String image;
+
+    protected Resources resources;
+
+    protected Probe livenessProbe;
+
+    protected Probe readinessProbe;
+
+    protected JvmOptions jvmOptions;
+
+    protected Map<String, Object> metrics;
+
+    protected Storage storage;
+
+    protected Affinity affinity;
+
+    @Description("The number of pods in the cluster.")
+    @Minimum(1)
+    @JsonProperty(required = true)
+    public int getReplicas() {
+        return replicas;
+    }
+
+    public void setReplicas(int replicas) {
+        this.replicas = replicas;
+    }
+
+    @Description("The docker image for the pods.")
+    @JsonInclude(JsonInclude.Include.NON_DEFAULT)
+    public String getImage() {
+        return image;
+    }
+
+    public void setImage(String image) {
+        this.image = image;
+    }
+
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    @Description("Resource constraints (limits and requests).")
+    public Resources getResources() {
+        return resources;
+    }
+
+    public void setResources(Resources resources) {
+        this.resources = resources;
+    }
+
+    @JsonInclude(JsonInclude.Include.NON_EMPTY)
+    @Description("Pod liveness checking.")
+    public Probe getLivenessProbe() {
+        return livenessProbe;
+    }
+
+    public void setLivenessProbe(Probe livenessProbe) {
+        this.livenessProbe = livenessProbe;
+    }
+
+    @JsonInclude(JsonInclude.Include.NON_DEFAULT)
+    @Description("Pod readiness checking.")
+    public Probe getReadinessProbe() {
+        return readinessProbe;
+    }
+
+    public void setReadinessProbe(Probe readinessProbe) {
+        this.readinessProbe = readinessProbe;
+    }
+
+    @JsonInclude(JsonInclude.Include.NON_EMPTY)
+    @Description("JVM Options for pods")
+    public JvmOptions getJvmOptions() {
+        return jvmOptions;
+    }
+
+    public void setJvmOptions(JvmOptions jvmOptions) {
+        this.jvmOptions = jvmOptions;
+    }
+
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    @Description("The Prometheus JMX Exporter configuration. " +
+            "See https://github.com/prometheus/jmx_exporter for details of the structure of this configuration.")
+    public Map<String, Object> getMetrics() {
+        return metrics;
+    }
+
+    public void setMetrics(Map<String, Object> metrics) {
+        this.metrics = metrics;
+    }
+
+    @Description("Persistent storage (disk). Cannot be updated.")
+    @JsonProperty(required = true)
+    public Storage getStorage() {
+        return storage;
+    }
+
+    public void setStorage(Storage storage) {
+        this.storage = storage;
+    }
+
+    @Description("Pod affinity rules.")
+    @KubeLink(group = "core", version = "v1", kind = "affinity")
```

Comment:
> do we need to be more specific here? Or is this really enough?

That, precisely, do you mean by 'this'? The `@KubeLink` or the `@Description`?

I used `@KubeLink` here because I'm reusing the Fabric8 (and hence K8S) `affinity`, so it's precisely the same. That's not the case with resources because we currently have our own classes for that (because we only support "memory" and "cpu", whereas in principle K8S can support other things too (i.e it's a map in k8s)). Of course we might want to revisit that decision and reuse the Fabric8 resources, but not in this PR.

Reply:
With _this_ I meant the `@KubeLink`. I'm a bit wondering if this specification is enough given that fields are added to the Kubernetes API without bumping the versions.